src/app/TargetSelect/TargetSelect.tsx

```diff
@@ -133,55 +109,45 @@ export const TargetSelect: React.FunctionComponent<TargetSelectProps> = (props)
       }
       setExpanded(false);
     },
-    [
-      context,
-      context.target,
-      selected,
-      notifications,
-      setExpanded,
-      removeCachedTargetSelection,
-      setCachedTargetSelection,
-    ]
+    [context.target, selected, notifications, setExpanded, removeCachedTargetSelection, setCachedTargetSelection]
   );
 
   const selectNone = React.useCallback(() => {
-    onSelect(undefined, undefined, true);
+    onSelect(undefined, NO_TARGET, true);
   }, [onSelect]);
 
+  const selectTargetFromCache = React.useCallback(
+    (targets) => {
+      if (targets.length) {
```

Comment:
I think this outer if/else can actually be removed. If the `targets` array is empty then the `targets.some()` call later will trivially return `false` and the same handling case runs either way.

```bash
$ node
> ['a'].some(v => true)
true
> [].some(v => true)
false
```

Reply:
Right! I will clean up this part too!